lib/iex/lib/iex/introspection.ex

```diff
@@ -39,112 +30,183 @@ defmodule IEx.Introspection do
-        nodocs("#{inspect module}.#{function}/#{arity}")
-    end
+    Enum.map(get_docs(module, function, arity),
+             fn {status, doc_list} -> display_doc_list(status, doc_list) end)
 
     dont_display_result
   end
 
-  defp h_mod_fun_arity(mod, fun, arity) when is_atom(mod) do
-    if docs = Code.get_docs(mod, :docs) do
-      doc = find_doc(docs, fun, arity, 4)
-            || find_default_doc(docs, fun, arity)
+  @doc """
+  Map over helpers and return a list of header, doc tuples.
+  """
+  def get_docs(module) do
+    helpers = IEx.Config.doc_helpers(:helpers)
+    opts = IEx.Config.doc_helpers(:find)
+    case opts do
+      :first ->
+        doc_list = Enum.find_value(helpers, fn(mod) -> can_help(mod, module) end)
+        case doc_list do
+          nil -> [{:not_found, {inspect(module),"No documentation found for #{inspect module}"}}]
+          _   -> doc_list
+        end
+      _ ->
+        helpers
+        |> Enum.map(fn(mod) -> mod.documentation(module) end)
+        |> Enum.filter(fn({status, _doc_list}) -> status != :unknown end)
+        # What about nil result
+    end
+  end
 
-      if doc do
-        print_doc(doc)
-        :ok
-      else
-        :not_found
-      end
-    else
-      :no_docs
+  @doc """
+  Map over helpers and return a list of header, doc tuples.
+  """
+  def get_docs(module, function) do
+    helpers = IEx.Config.doc_helpers(:helpers)
+    opts = IEx.Config.doc_helpers(:find)
+    case opts do
+      :first ->
+        doc_list = Enum.find_value(helpers, fn(mod) -> can_help(mod, module, function) end)
+        case doc_list do
+          nil -> [{:not_found, {inspect(module), "No documentation found for #{inspect module}"}}]
+          _   -> doc_list
+        end
+      _ ->
+        helpers
+        |> Enum.map(fn(mod) -> mod.documentation(module, function) end)
+        |> Enum.filter(fn({status, _doc_list}) -> status != :unknown end)
+        # What about nil result
+    end
+  end
+
+  @doc """
+  Map over helpers and return a list of header, doc tuples.
+  """
+  def get_docs(module, function, arity) do
+    helpers = IEx.Config.doc_helpers(:helpers)
+    opts = IEx.Config.doc_helpers(:find)
+    case opts do
+      :first ->
+        doc_list = Enum.find_value(helpers, fn(mod) -> can_help(mod, module, function, arity) end)
+        case doc_list do
+          nil -> [{:not_found, {inspect(module), "No documentation found for #{inspect module}"}}]
+          _   -> doc_list
+        end
+      _ ->
+        helpers
+        |> Enum.map(fn(mod) -> mod.documentation(module, function, arity) end)
+        |> Enum.filter(fn({status, _doc_list}) -> status != :unknown end)
+        # What about nil result
     end
   end
 
-  defp find_doc(docs, function, arity, pos) do
-    if doc = List.keyfind(docs, {function, arity}, 0) do
-      case elem(doc, pos) do
-        false -> nil
-        _ -> doc
-      end
+  @doc """
+  Return nil if mod.documentation returns :unknown
```

Comment:
verbs should in in present continuos, not in imperative form. it should read.
> Returns `nil` if mod.documentation returns :unknown

nil should have backticks, and I'm not sure if we should have backticks in `mod.documentation` (probably yes)